packages/daemon/test/test-endo.js

```diff
@@ -1363,6 +1366,37 @@ test('read remote value', async t => {
   t.is(hostAValue, 'hello, world!');
 });
 
+test('round-trip remotable identity', async t => {
+  // Also called grant matching.
+  const hostA = await prepareHostWithTestNetwork(t);
+  const hostB = await prepareHostWithTestNetwork(t);
+
+  // Introduce A to B (allow B to infer A)
+  await E(hostA).addPeerInfo(await E(hostB).getPeerInfo());
+
+  await E(hostB).evaluate(
+    'MAIN',
+    'Far("Echoer", { echo: value => value })',
+    [],
+    [],
+    'echoer',
+  );
+  const echoerId = await E(hostB).identify('echoer');
+  await E(hostA).write(['echoer'], echoerId);
+  const survivedEcho = await E(hostA).evaluate(
+    'MAIN',
+    `
+      const token = Far('Token', {});
+      E(echoer).echo(token).then(allegedlyIdenticalToken =>
+        token === allegedlyIdenticalToken
+      );
+    `,
+    ['echoer'],
+    ['echoer'],
+  );
+  t.assert(survivedEcho);
+});
```

Comment:
This test fails on master because master does not attempt to deduplicate the A->B and B<-A CapTP connections.

Reply:
I was mistaken. This was passing. I created a more specific test that exercised a hello from the remote and integrated the remote control into the provider.